src/main/java/net/sf/jabref/importer/fileformat/MedlineImporter.java

```diff
@@ -84,50 +121,371 @@ public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
+            }
+
+        }
+
+        BibEntry entry = new BibEntry(IdGenerator.next(), "article"); // id assumes an existing database so don't create one here
+        entry.setField(fields);
+
+        bibItems.add(entry);
+    }
+
+    private void addNotes(HashMap<String, String> fields, List<GeneralNote> generalNote) {
+        List<String> notes = new ArrayList<>();
+        for (GeneralNote note : generalNote) {
+            if (note != null) {
+                notes.add(note.getContent());
+            }
+        }
+        fields.put("note", join(notes, ", "));
+    }
+
+    private void addInvestigators(HashMap<String, String> fields, InvestigatorList investigatorList) {
+        List<String> investigatorNames = new ArrayList<>();
+        List<String> affiliationInfos = new ArrayList<>();
+        String name = "";
+        // add the investigators like the authors
+        if (investigatorList.getInvestigator() != null) {
+            List<Investigator> investigators = investigatorList.getInvestigator();
+            for (Investigator investigator : investigators) {
+                name = investigator.getLastName();
+                if (investigator.getForeName() != null) {
+                    name += ", " + investigator.getForeName();
+                }
+                investigatorNames.add(name);
+
+                //now add the affiliation info
+                if (investigator.getAffiliationInfo() != null) {
+                    for (AffiliationInfo info : investigator.getAffiliationInfo()) {
+                        for (Serializable affiliation : info.getAffiliation().getContent()) {
+                            if (affiliation instanceof String) {
+                                affiliationInfos.add((String) affiliation);
+                            }
+                        }
+                    }
+                    fields.put("affiliation", join(affiliationInfos, ", "));
+                }
+            }
+            fields.put("investigator", join(investigatorNames, " and "));
+        }
+    }
+
+    private void addKeyWords(HashMap<String, String> fields, List<KeywordList> allKeywordLists) {
+        List<String> keywordStrings = new ArrayList<>();
+        //add keywords to the list
+        for (KeywordList keywordList : allKeywordLists) {
+            for (Keyword keyword : keywordList.getKeyword()) {
+                for (Serializable content : keyword.getContent()) {
+                    if (content instanceof String) {
+                        keywordStrings.add((String) content);
+                    }
+                }
+            }
+        }
+        //Check whether MeshHeadingList exist or not
+        if (fields.get("keywords") == null) {
+            fields.put("keywords", join(keywordStrings, "; "));
+        } else {
+            if (keywordStrings.size() > 0) {
+                //if it exists, combine the MeshHeading with the keywords
+                String result = join(keywordStrings, "; ");
```

Comment:
Keywords should be joined with the configured at `net.sf.jabref.JabRefPreferences.KEYWORD_SEPARATOR`.